File: problems/643-maximum-average-subarray-i/maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
const findMaxAverage = (nums, k) => {
    let sum = 0;
    for (let i = 0; i < k; i++) {
        sum += nums[i]
    }
    let maxSum = sum;
    for (let i = 0; i < nums.length - k; i++) {
        sum -= nums[i];
        sum += nums[i + k];
        maxSum = Math.max(maxSum, sum)
    }
    return maxSum / k
};

File: problems/643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxSum = sum(nums[:k])
        tmpSum = maxSum
        for i in range(len(nums) - k):
            tmpSum -= nums[i]
            tmpSum += nums[i + k]
            maxSum = max(tmpSum, maxSum)
        return maxSum / k